src/cuga/backend/server/manage_routes/knowledge_routes.py

```
@@ -0,0 +1,464 @@
+        merged = {**existing_knowledge, **knowledge}
+
+        known_fields = {f.name for f in dc_fields(KnowledgeConfig)} - {"persist_dir"}
+        filtered = {k: v for k, v in merged.items() if k in known_fields}
+
+        prev_adapt_hash = client_adaptation_hash(existing_knowledge.get("client_adaptation_text", ""))
+        _prev_gloss_hash = client_glossary_hash(existing_knowledge.get("client_adaptation_glossary", []))
+
+        try:
+            validated = KnowledgeConfig.coerce_and_validate(filtered)
+        except ClientAdaptationError as cae:
+            raise HTTPException(status_code=422, detail=cae.to_dict())
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)